src/main/java/org/myrobotlab/programab/PredicateEvent.java

```diff
@@ -0,0 +1,29 @@
+package org.myrobotlab.programab;
+
+/**
+ * Pojo for state change of one of ProgramAB's state info
+ * @author GroG
+ *
+ */
+public class PredicateEvent {
+  /**
+   * unique identifier for the session user & bot
+   */
+  public String id;
```

Comment:
session id should be decomposed to the bot and the user...  this id is compound and could change in the future.

Reply:
updated